### PR TITLE
chore : Fix vulkan related compiler warnings, add help text, improve CLI options

### PR DESCRIPTION
**Pull request by teleprint-me**
# Fix vulkan related compiler warnings, add help text, improve CLI options

## Overview
- Add prototypes for function definitions
- Invert logic of --no-clean option to be more intuitive
- Provide a new help prompt with clear instructions

### Details
This PR addresses several issues related to the CLI tool for generating Vulkan shaders. The main changes include adding prototypes for function definitions, inverting the logic of the `--no-clean` option, and providing a new help prompt with clear instructions on how to use the tool effectively.

### Notes
The addition of prototypes ensures that there are no undefined references during compilation. Inverting the logic of the --no-clean option makes it more intuitive for users to specify whether or not they want to remove temporary files after building. The new help prompt provides clear instructions on how to use the tool, making it easier for others to adopt and understand its capabilities.

## Summary
- [x] I have read the [contributing guidelines](https://github.com/ggerganov/llama.cpp/blob/master/CONTRIBUTING.md)
- Self-reported review complexity:
  - [x] Low
  - [ ] Medium
  - [ ] High

I've tested this PR locally and ensured that all changes are functional and do not introduce any regressions in existing functionality.

```sh
$ ctest --rerun-failed --output-on-failure
Test project /mnt/valerie/forked/ggerganov/llama.cpp/build
# ...
100% tests passed, 0 tests failed out of 25

Label Time Summary:
curl             =   0.61 sec*proc (1 test)
eval-callback    =   0.61 sec*proc (1 test)
main             =  48.64 sec*proc (22 tests)
model            =   0.01 sec*proc (2 tests)

Total Test time (real) =  49.28 sec
```

Please review my changes and let me know if there are any questions or concerns before merging!

### CODE REVIEW

**Comment by teleprint-me** [comment]
I didn't notice that the binary wasn't ignored until afterwards. It's a minor quality of life change. Hopefully it's okay to bulk it in with this PR.

File: ggml/src/vulkan-shaders/vulkan-shaders-gen.cpp [hunk 1]
```diff
@@ -481,6 +496,18 @@ int main(int argc, char** argv) {
         }
     }
 
+    if (argc <= 1 || std::string(argv[1]) == "--help") {
```
Comment (0cc4m) [inline ggml/src/vulkan-shaders/vulkan-shaders-gen.cpp:499]:
```suggestion
    if (argc <= 1 || args.find("--help") != args.end()) {
```
This is nitpicking, but it's nice to be able to just put --help at the end of a longer command.

File: ggml/src/vulkan-shaders/vulkan-shaders-gen.cpp [hunk 2]
```diff
@@ -426,7 +440,7 @@ void process_shaders(std::vector<std::future<void>>& tasks) {
     }));
 }
 
-void write_output_files() {
+void write_output_files(void) {
```
Comment (0cc4m) [inline ggml/src/vulkan-shaders/vulkan-shaders-gen.cpp:443]: Why would not having `void` in the empty parameter list cause a compiler warning? AFAIK that's required in C, but not C++.

**Review by 0cc4m — APPROVED** [review]
Just minor comments, looks good and works fine on Linux.

File: ggml/src/vulkan-shaders/vulkan-shaders-gen.cpp (hunk 2 above)
Comment (teleprint-me) [reply ggml/src/vulkan-shaders/vulkan-shaders-gen.cpp:443]: C was my "first" real programming language and it's mostly out of habit.

<details>
<summary>here's why by example</summary>

```sh
21:50:05 | ~/Local/code/c
  λ cat test.c 
#include <stdio.h>

void print_text();

void print_text() {
    fprintf(stderr, "Hello, world!\n");
}

int main() {
    print_text();
    return 0;
}
21:50:09 | ~/Local/code/c
  λ gcc -Wall -Wextra -std=c89 -pedantic -Wmissing-prototypes -Wstrict-prototypes test.c -o test
test.c:3:1: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
    3 | void print_text();
      | ^~~~
test.c:5:6: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
    5 | void print_text() {
      |      ^~~~~~~~~~
test.c:9:5: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
    9 | int main() {
      |     ^~~~
21:50:13 | ~/Local/code/c
  λ vim test.c
21:51:23 | ~/Local/code/c
  λ cat test.c
#include <stdio.h>

void print_text(void);

void print_text(void) {
    fprintf(stderr, "Hello, world!\n");
}

int main(void) {
    print_text();
    return 0;
}
21:51:25 | ~/Local/code/c
  λ gcc -Wall -Wextra -std=c89 -pedantic -Wmissing-prototypes -Wstrict-prototypes test.c -o test
21:51:28 | ~/Local/code/c
  λ ./test 
Hello, world!
```
</details>

Comment (0cc4m) [reply ggml/src/vulkan-shaders/vulkan-shaders-gen.cpp:443]: Strict prototypes definitely make sense in C, but in C++ this is already enforced as part of the language: an empty parameter list does actually mean no parameters. For that reason I would rather leave out the `void` here.

Comment (teleprint-me) [reply ggml/src/vulkan-shaders/vulkan-shaders-gen.cpp]: Ok. Done.

Comment (0cc4m) [reply ggml/src/vulkan-shaders/vulkan-shaders-gen.cpp]: Thank you.

**Pull request merged**